helm/botkube/values.yaml

```diff
@@ -7,6 +7,7 @@ replicaCount: 1
 image:
   repository: infracloud/botkube
   pullPolicy: Always
+  # tag: latest
```

Comment:
We should keep this line uncommented. What is the default value we are considering currently?

Reply:
Currently the default value is `{{ .Chart.Version }}` in deployment.yaml

Reply:
@PrasadG193 ^

Reply:
@bhavin192 if we keep it blank, it should work. We should be able to set tag with `--set image.tag=latest` without making any change in the chart. Can you please check if the variable needs to be present in values.yaml if we want to set it with `--set` command?

Reply:
>  if we keep it blank

Can you please elaborate this? Blank as in commented or like `tag: ""`?
> without making any change in the chart.

Do you mean without making changes to deployment.yaml? 

And yes, it works even if you don't have value in values.yaml but keeping it commented in `values.yam` would make it easy for people to figure out that it's possible to set tags as well.

Reply:
> Can you please elaborate this? Blank as in commented or like tag: ""?
Yes, or something like `tag:  `. I think, commenting field will create confusion

Reply:
I feel blank field will create confusion :smile:, how about adding some explanation like: 
```yaml
# default tag is Chart version which is picked from Charts.yaml if you
# want to use some other tag it can be specified here
```

Reply:
yes, that sounds good

Reply:
Helm best practices recommend to use in `Chart.yaml` `appVersion:` to set application version.
Then it can be used in the deployment as: 
```
image: "{{ .Values.image.repository }}::{{ default .Chart.AppVersion .Values.image.tag }}"
```

Reply:
And the chart version should be updated only when there is a change in the chart itself not the application, that's what I have seen with helm stable repo.

Reply:
no, if you update appversion, you need to update chat version as well!
or if appversion stays the same, chart gets changes, you update chat version.
Any changes to the chart you bump chart version :)

Reply:
Yep, that's what I meant to say but missed the fact that changing Charts.yaml means changing chart version as well.

Reply:
👍

Reply:
@rimusz can I use this https://github.com/infracloudio/botkube/pull/89#discussion_r282802760 in my commit?

Reply:
sure